### PR TITLE
[REF] lru-cache implementation

### PR DESCRIPTION
**Pull request by moylop260**
Test reducing from 20 seconds to 10 seconds generate a CFDI invoice:

[IMAGE ALT TEXT HERE](https://www.youtube.com/watch?v=3aJn4j66g54)

With last commit the time are 2 seconds:
[IMAGE ALT TEXT HERE](https://www.youtube.com/watch?v=INbwwhyHRa0)

Fix: https://github.com/Vauxoo/cfdilib/issues/47

### CODE REVIEW

File: cfdilib/tools.py [hunk 1]
```diff
@@ -62,6 +69,27 @@ def f_retry(*args, **kwargs):
     return deco_retry
 
 
+@lru_cache(maxsize=128)
+@retry(urllib2.URLError, tries=3, delay=3, back_off=2)
+def _cache_it(url):
```
Comment (nhomar) [inline cfdilib/tools.py]: change this logic a little to return early and I will be a happy guy.

Comment (moylop260) [inline cfdilib/tools.py]: done

Comment (moylop260) [inline cfdilib/tools.py]: Wait I'm testing something...

Comment (moylop260) [inline cfdilib/tools.py]: Tested!
This pr is ready

**Pull request merged**